### PR TITLE
Forgot password

### PR DESCRIPTION
**Pull request by nguyenvothuan**
# Forgot password API.
## Description:
- Let me explain from the client's side. So when the user click "Forgot password" on the client, the client will send the user's email along with url to the reset password page of the client to a POST at `api/accounts/forgot-password`. The server receives these parameters, and return 200 no matter what. If the user exists in the db, the server will send an email with the url provided by the client, along with the access token as a parameter in this url, to the user's email. The user will then click on this url in his email, and access the reset password page on the client's side.
- The client is now responsible for extracting this token, putting it in the headers, and prompts the user to enter a new password. It then sends the new password to a PUT API: `/api/accounts/reset-password`. If the token is valid, reset the password with `set_password` (you don't have to worry if the password is exposed since `set_password` will encrypt it before saving to the db). Return 403 otherwise. See issue #21 for more detail. 
## Demo:
https://www.dropbox.com/s/ztlmut3ft10qr93/Django%20client%20base%20URL%20-%20Google%20Chrome%202023-01-11%2010-42-18%20%281%29.mp4?dl=0

### CODE REVIEW

**Comment by kien-to** [comment]
Quay cai video ChatGPT lam gi?

**Comment by nguyenvothuan** [comment]
ua wtf doi ti

**Comment by kien-to** [comment]
The instruction says access a link to reset password, but you are using the token to reset password. Customer will be confused.

File: accounts/models.py [hunk 1]
```diff
@@ -58,17 +61,17 @@ class User(AbstractBaseUser, PermissionsMixin):
     updated = AutoDateTimeField(default=timezone.now)
     # if a student account
     is_student = models.BooleanField(default=False)
-    student = models.OneToOneField(Student, on_delete=models.CASCADE,  null=True,default=None)
-    #if an recruiter account
+    student = models.OneToOneField(
+        Student, on_delete=models.CASCADE,  null=True, default=None)
```
Comment (theanh28) [inline accounts/models.py:65]: ",  null=True" -> double space

**Comment by theanh28** [comment]
The rest LGTM!

**Comment by nguyenvothuan** [comment]
@kien-to the email sent contains the url to the reset password page on the client-side. The client has to pass in its url to that reset-password page, for example 'localhost:3000/reset-password', then the server will generate an access token and append it to this 'localhost:3000/reset-password', something like 'localhost:3000/reset-password?accesstoken=123abcasd'. The user click on this url in his email inbox, and access the reset password page, with the access token already in its header. Hiểu ko?

**Comment by theanh28** [comment]
> @kien-to the email sent contains the url to the reset password page on the client-side. The client has to pass in its url to that reset-password page, for example 'localhost:3000/reset-password', then the server will generate an access token and append it to this 'localhost:3000/reset-password', something like 'localhost:3000/reset-password?accesstoken=123abcasd'. The user click on this url in his email inbox, and access the reset password page, with the access token already in its header. Hiểu ko?

Hmmmmm, m có nghĩ server nên là ng tạo ra url chứ ko phải client gửi vè ko. Ở bất kì env nào, server cũng nên biết url của client/wrbsite mà nó sẽ serve (e.g. dev: localhost:3000, qa: qaurl something, prod: proj-edu.io), nên để server lo chuyện đó thì hay hơn.

**Comment by nguyenvothuan** [comment]
> 

Good idea, I added a config file to store these constants.

**Pull request merged**